src/lib/optimizer/join_ordering/enumerate_ccp.cpp

```diff
@@ -137,9 +137,12 @@ void EnumerateCcp::_enumerate_cmp(const JoinGraphVertexSet& primary_vertex_set)
 
     _csg_cmp_pairs.emplace_back(primary_vertex_set, cmp_vertex_set);
 
+    // In the original paper, the extended exclusion set is the union of the exclusion set and the neighborhood.
+    // However, we use the corrected version fron the errata: union of the exclusion set and the exclusion set of the
+    // neighborhood.
     const auto extended_exclusion_set = exclusion_set | (_exclusion_set(*iter) & neighborhood);
 
-    std::vector<JoinGraphVertexSet> csgs;
+    auto csgs = std::vector<JoinGraphVertexSet>{};
```

Comment:
Possible to reserve?

Reply:
Looking at the code in the called methods: I do not think so. Seems to heavily depend on the concrete join graph.

Reply:
Did the following:
https://github.com/hyrise/hyrise/blob/bd6c2e649fa397ecae38370f97619fbe35c2650d/src/bin/playground.cpp#L15-L57

The second version was faster on Mac and nemea. Thus, let's use a single bitset with `reset`.